### DISCUSSION TITLE
Automatic Savings Transaction

### DISCUSSION THREAD

**Discussion by isaacgr** [firefly-iii/firefly-iii#6866] [Firefly III] [answered]
Hi,

My chequing account is setup to deposit $0.50 into my savings account for every transaction I make with my debit card. Currently I don't see any way to automatically configure this via a rule (so I dont have to constantly add those small transfers manually). Am I missing something?

**Comment by JC5** [comment, marked as answer]
There is no option to do this automatically, but there is a script that can do this for you. 

https://github.com/jc5/autosave

**Reply by isaacgr** [reply]
Thanks for the script. I'll check it out.

Would you be averse to adding this feature?

**Reply by JC5** [reply]
Yes, that's why I made the script 😉

**Reply by Doddophonique** [reply]
> There is no option to do this automatically, but there is a script that can do this for you.
> 
> https://github.com/jc5/autosave

I'm trying the script right now, but it fails at [line 246](https://github.com/JC5/autosave/blob/main/autosave.php#L246).

It looks like something has changed in the API, as I can't find that endpoint [in the API docs](https://api-docs.firefly-iii.org/?urls.primaryName=1.5.6).
```
Request /api/v1/transactions/7/transaction_links returned with HTTP code 404.

{"message":"Resource not found","exception":"NotFoundHttpException"}
```

Any suggestions on this? I tried a couple of modification by intuition (not really a PHP expert) but it has not been successful.

**Reply by JC5** [reply]
No I'll check it out, thanks for letting me know!

**Reply by JC5** [reply]
Script should work again!

**Reply by isaacgr** [reply]
Thanks!

Just curious. Why the aversion to adding something like this as a rule (rules which can setup transfers or transactions could be pretty powerful)? I have an additional user that isn't a programmer who would be unable to manage this, leaving me to do it in some form.
